certif/tests/unit/services/session_test.js

```diff
@@ -0,0 +1,197 @@
+    sessionService.currentDomain = { getExtension: sinon.stub() };
+    sessionService.currentUser = { load: sinon.stub(), certificationPointOfContact: null };
+
+    routerService = this.owner.lookup('service:router');
+    routerService.transitionTo = sinon.stub();
+  });
+
+  module('#handleAuthentication', function () {
+    test('calls loadCurrentUserAndSetLocale with no arg', async function (assert) {
+      // given
+      sessionService.loadCurrentUserAndSetLocale = sinon.stub();
+      sessionService.currentUser = { certificationPointOfContact: { isMemberOfACertificationCenter: false } };
+      sessionService._setLocale = sinon.stub();
+
+      // when
+      await sessionService.handleAuthentication();
+
+      // then
+      sinon.assert.calledWith(sessionService.loadCurrentUserAndSetLocale);
+      assert.ok(true);
+    });
+  });
+
+  module('#loadCurrentUserAndSetLocale', function () {
+    module('when domain is .fr', function () {
+      module('when no lang query parameter', function () {
+        module('when user is not loaded', function () {
+          test('calls _setLocale with fr', async function (assert) {
+            // given
+            sessionService.currentDomain.getExtension.returns('fr');
+            sessionService._setLocale = sinon.stub();
+
+            // when
+            await sessionService.loadCurrentUserAndSetLocale();
+
+            // then
+            sinon.assert.calledWith(sessionService._setLocale, 'fr');
+            assert.ok(true);
+          });
+        });
+
+        module('when user is loaded', function () {
+          test('calls _setLocale with fr', async function (assert) {
+            // given
+            sessionService.currentDomain.getExtension.returns('fr');
+            sessionService.currentUser.certificationPointOfContact = {
+              lang: 'user lang has no impact in this scenario',
+            };
+            sessionService._setLocale = sinon.stub();
+
+            // when
+            await sessionService.loadCurrentUserAndSetLocale();
+
+            // then
+            sinon.assert.calledWith(sessionService._setLocale, 'fr');
+            assert.ok(true);
+          });
+        });
+      });
+      module('when lang query parameter', function () {
+        module('when user is not loaded', function () {
+          test('calls _setLocale with fr', async function (assert) {
+            // given
+            sessionService.currentDomain.getExtension.returns('fr');
+            sessionService._setLocale = sinon.stub();
+
+            // when
+            const localeFromQueryParam = 'en';
+            await sessionService.loadCurrentUserAndSetLocale(localeFromQueryParam);
+
+            // then
+            sinon.assert.calledWith(sessionService._setLocale, 'fr');
+            assert.ok(true);
+          });
+        });
+
+        module('when user is loaded', function () {
+          test('calls _setLocale with fr', async function (assert) {
```

Comment:
Même genre de remarque que plus haut, on a l'impression que user is load est une condition très importante alors que c'est en terme de priorité la troisième façon de setter la locale

Reply:
Les intitulés des tests ont été modifiés pour être « métier » et non plus « technique ».